### ISSUE TITLE
Unable to properly delete uploaded artifacts ("Nothing to upload, all files are on the server")

### ISSUE THREAD

**Issue by jenskuhrjorgensen** [getsentry/sentry-cli#1724]
### Issue

If I delete all uploaded artifact bundles on sentry.io and then try to upload the same artifacts (i.e. without making any changes that would produce a new bundle/"Bundle ID") I get "Nothing to upload, all files are on the server". However, my uploaded events are not being retraced due to "Source Maps not uploaded ", so the deleted artifacts are only partially there, i.e. they are blocking me from uploading new ones for the same bundle but cannot be inspected or deleted or used for retracing. That is pretty confusing. 

### Environment

sentry-cli version 1.75.2
MacOS

### Steps to Reproduce

1. `npx react-native@latest init SentryTest`
2. `sentry-wizard -i reactNative -p android`
3. Answer the four prompts (organization slug, project slug, DSN, auth token)
4. `npx react-native run-android --mode release`
5. Delete all uploaded artifact bundles from sentry.io https://test-32ecaf969.sentry.io/settings/projects/sentry-test/source-maps/artifact-bundles/
![Screenshot 2023-08-29 at 14 21 44](https://github.com/getsentry/sentry-react-native/assets/28528363/5667cd7d-7744-4953-98df-d1f32ce39f89)
6. Rerun `npx react-native run-android --mode release`

### Actual Result

Nothing is uploaded even though there are no artifact (or release) bundles present in sentry.io:

```
> Task :app:createBundleReleaseJsAndAssets_SentryUpload_com.sentrytest@1.0+1_1
Processing react-native sourcemaps for Sentry upload.
> Analyzing 2 sources
> Rewriting sources
> Adding source map references
Uploading sourcemaps for release com.sentrytest@1.0+1 distribution 1
> Bundled 2 files for upload
> Bundle ID: dea28e39-4391-56d5-8879-f0b9dc30a906
> Nothing to upload, all files are on the server
```

### Expected result

I would expect that deleting the artifact bundles from sentry.io would allow me to upload them again. However, it looks like the artifacts are somewhat still present somewhere where I cannot properly delete them from.

**Comment by loewenheim** [comment]
Thank you for the report, I will have a look at this ASAP.

**Comment by loewenheim** [comment]
Can you try to run it again with `--log-level debug`?

**Comment by jenskuhrjorgensen** [comment]
Sure, below output is for this flow of actions: 
1. `SENTRY_LOG_LEVEL=debug npx react-native run-android --mode release`
2. Manually delete all artifacts from sentry.io
3. Rerun `SENTRY_LOG_LEVEL=debug npx react-native run-android --mode release`

```
sentry-test % SENTRY_LOG_LEVEL=debug npx react-native run-android --mode release
info JS server already running.
info Installing the app...

> Task :sentry_react-native:compileReleaseJavaWithJavac

> Task :app:createBundleReleaseJsAndAssets
debug Reading Metro config from /Users/<user>/Documents/repos/sentry-test/metro.config.js
warning: the transform cache was reset.
                Welcome to Metro v0.76.8
              Fast - Scalable - Integrated


info Writing bundle output to:, /Users/<user>/Documents/repos/sentry-test/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle
info Writing sourcemap output to:, /Users/<user>/Documents/repos/sentry-test/android/app/build/intermediates/sourcemaps/react/release/index.android.bundle.packager.map
info Done writing bundle output
info Done writing sourcemap output
info Copying 1 asset files
info Done copying assets

> Task :app:createBundleReleaseJsAndAssets_SentryUpload_com.sentrytest@1.0+1_1
Processing react-native sourcemaps for Sentry upload.
> Analyzing 2 sources
> Rewriting sources
> Adding source map references
Uploading sourcemaps for release com.sentrytest@1.0+1 distribution 1
> Bundled 2 files for upload
> Bundle ID: 8f5c3b9d-2399-55fc-9424-9ab88438daaa
> Uploaded files to Sentry
> File upload complete (processing pending on server)
> Organization: test-32ecaf969
> Project: sentry-test
> Release: com.sentrytest@1.0+1
> Dist: 1
> Upload type: artifact bundle

Source Map Upload Report
  Minified Scripts
    ~/index.android.bundle (sourcemap at index.android.bundle.map)
  Source Maps
    ~/index.android.bundle.map

> Task :app:createBundleReleaseJsAndAssets_SentryCollectModules_com.sentrytest@1.0+1_1
Sentry Logger [info]: Reading source map from /Users/<user>/Documents/repos/sentry-test/android/app/build/generated/sourcemaps/react/release/index.android.bundle.map
Sentry Logger [info]: Saving modules to /Users/<user>/Documents/repos/sentry-test/android/app/src/main/assets/modules.json
Sentry Logger [info]: Resolving modules from paths /Users/<user>/Documents/repos/sentry-test/android/app/src/main/assets/modules.json
Sentry Logger [info]: Modules collected and saved to: /Users/<user>/Documents/repos/sentry-test/android/app/src/main/assets/modules.json

> Task :app:installRelease
Installing APK 'app-release.apk' on '1Pixel_2_API_30(AVD) - 11' for :app:release
Installed on 1 device.

BUILD SUCCESSFUL in 46s
66 actionable tasks: 35 executed, 31 up-to-date
info Connecting to the development server...
info Starting the app on "emulator-5554"...
Starting: Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] cmp=com.sentrytest/.MainActivity }
sentry-test % SENTRY_LOG_LEVEL=debug npx react-native run-android --mode release
info JS server already running.
info Installing the app...

> Task :app:createBundleReleaseJsAndAssets_SentryUpload_com.sentrytest@1.0+1_1
Processing react-native sourcemaps for Sentry upload.
> Analyzing 2 sources
> Rewriting sources
> Adding source map references
Uploading sourcemaps for release com.sentrytest@1.0+1 distribution 1
> Bundled 2 files for upload
> Bundle ID: 8f5c3b9d-2399-55fc-9424-9ab88438daaa
> Nothing to upload, all files are on the server

Source Map Upload Report
  Minified Scripts
    ~/index.android.bundle (sourcemap at index.android.bundle.map)
  Source Maps
    ~/index.android.bundle.map

> Task :app:createBundleReleaseJsAndAssets_SentryCollectModules_com.sentrytest@1.0+1_1
Sentry Logger [info]: Reading source map from /Users/<user>/Documents/repos/sentry-test/android/app/build/generated/sourcemaps/react/release/index.android.bundle.map
Sentry Logger [info]: Saving modules to /Users/<user>/Documents/repos/sentry-test/android/app/src/main/assets/modules.json
Sentry Logger [info]: Resolving modules from paths /Users/<user>/Documents/repos/sentry-test/android/app/src/main/assets/modules.json
Sentry Logger [info]: Modules collected and saved to: /Users/<user>/Documents/repos/sentry-test/android/app/src/main/assets/modules.json

> Task :app:installRelease
Installing APK 'app-release.apk' on '1Pixel_2_API_30(AVD) - 11' for :app:release
Installed on 1 device.

BUILD SUCCESSFUL in 7s
66 actionable tasks: 6 executed, 60 up-to-date
info Connecting to the development server...
info Starting the app on "emulator-5554"...
Starting: Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] cmp=com.sentrytest/.MainActivity }
sentry-test %
```

**Comment by Swatinem** [comment]
Your log seems to invoke upload twice, using the same bundle, so the second time around, nothing is uploaded as the first one is already being saved/assembled on the server side.

```
> Task :app:createBundleReleaseJsAndAssets_SentryUpload_com.sentrytest@1.0+1_1
Processing react-native sourcemaps for Sentry upload.
> Analyzing 2 sources
> Rewriting sources
> Adding source map references
Uploading sourcemaps for release com.sentrytest@1.0+1 distribution 1
> Bundled 2 files for upload
> Bundle ID: 8f5c3b9d-2399-55fc-9424-9ab88438daaa
> Uploaded files to Sentry
> File upload complete (processing pending on server)

[…]

> Task :app:createBundleReleaseJsAndAssets_SentryUpload_com.sentrytest@1.0+1_1
Processing react-native sourcemaps for Sentry upload.
> Analyzing 2 sources
> Rewriting sources
> Adding source map references
Uploading sourcemaps for release com.sentrytest@1.0+1 distribution 1
> Bundled 2 files for upload
> Bundle ID: 8f5c3b9d-2399-55fc-9424-9ab88438daaa
> Nothing to upload, all files are on the server
```

We are currently investigating an issue where the upload status (called "assemble status" internally) is being persisted for 10 minutes after an upload, and re-uploading within that timeframe will report an already existing file.

**Comment by loewenheim** [comment]
See https://github.com/getsentry/sentry/pull/55455.

**Comment by loewenheim** [comment]
Sorry, closed this prematurely. I can't reproduce the issue locally anymore with the changes to `sentry-cli` and `sentry`, but the `sentry-cli` fix is not yet released.

**Comment by jenskuhrjorgensen** [comment]
@loewenheim sorry, I'm not sure what that means. Does it mean that you made a fix that is yet to be released in a new version if the cli? :)

**Comment by loewenheim** [comment]
Yes, exactly. We're going to have the fix out in the next few days.

**Comment by jenskuhrjorgensen** [comment]
@loewenheim this issue is still present in the newest version of sentry-cli (2.21.1): When I delete all uploaded "Artifact Bundles" from our project in sentry.io and rerun the upload command it says:
> Nothing to upload, all files are on the server

However, a new artifact bundle becomes visible in sentry.io after running the command, so something was either uploaded or maybe picked up from some backend cache instead of using the newly built one that I tried to upload (maybe because the bundle id was the same as some previously uploaded and "deleted" bundle). 

It can be reproduced in a vanilla React Native project with Sentry added via the Sentry Wizard. I made a small repro (contains some other changes too, because I also used it for reporting another issue):
https://github.com/jenskuhrjorgensen/sentry-test-logger

Debug log:

```
android % SENTRY_LOG_LEVEL=debug ./gradlew installRelease

> Task :app:createBundleReleaseJsAndAssets_SentryUpload_com.sentrytestlogger@1.0+1_1
  INFO    2023-10-02 11:41:02.917820 +02:00 Loaded file referenced by SENTRY_PROPERTIES (/Users/johndoe/Documents/repos/sentry-test-logger/android/sentry.properties)
  INFO    2023-10-02 11:41:02.930418 +02:00 Loaded config from /Users/johndoe/.sentryclirc
  DEBUG   2023-10-02 11:41:02.930435 +02:00 sentry-cli version: 2.21.1, platform: "darwin", architecture: "x86_64"
  INFO    2023-10-02 11:41:02.930516 +02:00 sentry-cli was invoked with the following command line: "/Users/johndoe/Documents/repos/sentry-test-logger/node_modules/@sentry/cli/sentry-cli" "react-native" "gradle" "--bundle" "/Users/johndoe/Documents/repos/sentry-test-logger/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle" "--sourcemap" "/Users/johndoe/Documents/repos/sentry-test-logger/android/app/build/generated/sourcemaps/react/release/index.android.bundle.map" "--release" "com.sentrytestlogger@1.0+1" "--dist" "1"
  INFO    2023-10-02 11:41:02.931918 +02:00 Issuing a command for Organization: test-32ecaf969 Project: sentry-test
Processing react-native sourcemaps for Sentry upload.
  INFO    2023-10-02 11:41:02.931944 +02:00   bundle path: /Users/johndoe/Documents/repos/sentry-test-logger/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle
  INFO    2023-10-02 11:41:02.931959 +02:00   sourcemap path: /Users/johndoe/Documents/repos/sentry-test-logger/android/app/build/generated/sourcemaps/react/release/index.android.bundle.map
  DEBUG   2023-10-02 11:41:02.940621 +02:00 Non-file bundle found
> Analyzing 2 sources
> Rewriting sources
  DEBUG   2023-10-02 11:41:03.220501 +02:00 > No debug id found for ~/index.android.bundle.map to reference
> Adding source map references
  DEBUG   2023-10-02 11:41:03.221485 +02:00 request GET https://sentry.io/api/0/organizations/test-32ecaf969/chunk-upload/
  DEBUG   2023-10-02 11:41:03.221521 +02:00 using token authentication
  DEBUG   2023-10-02 11:41:03.221537 +02:00 retry number 0, max retries: 0
  DEBUG   2023-10-02 11:41:03.423973 +02:00 > GET /api/0/organizations/test-32ecaf969/chunk-upload/ HTTP/1.1
  DEBUG   2023-10-02 11:41:03.423993 +02:00 > Host: sentry.io
  DEBUG   2023-10-02 11:41:03.423999 +02:00 > Accept: */*
  DEBUG   2023-10-02 11:41:03.424004 +02:00 > Connection: TE
  DEBUG   2023-10-02 11:41:03.424008 +02:00 > TE: gzip
  DEBUG   2023-10-02 11:41:03.424013 +02:00 > User-Agent: sentry-cli/2.21.1
  DEBUG   2023-10-02 11:41:03.424481 +02:00 > Authorization: Bearer sntrys_e***
  DEBUG   2023-10-02 11:41:03.602068 +02:00 < HTTP/1.1 200 OK
  DEBUG   2023-10-02 11:41:03.602099 +02:00 < server: nginx
  DEBUG   2023-10-02 11:41:03.602113 +02:00 < date: Mon, 02 Oct 2023 09:41:03 GMT
  DEBUG   2023-10-02 11:41:03.602133 +02:00 < content-type: application/json
  DEBUG   2023-10-02 11:41:03.602169 +02:00 < Content-Length: 345
  DEBUG   2023-10-02 11:41:03.602192 +02:00 < allow: GET, POST, HEAD, OPTIONS
Uploading sourcemaps for release com.sentrytestlogger@1.0+1 distribution 1
  DEBUG   2023-10-02 11:41:03.602203 +02:00 < access-control-allow-methods: GET, POST, HEAD, OPTIONS
  DEBUG   2023-10-02 11:41:03.602222 +02:00 < access-control-allow-headers: X-Sentry-Auth, X-Requested-With, Origin, Accept, Content-Type, Authentication, Authorization, Content-Encoding, sentry-trace, baggage, X-CSRFToken
  DEBUG   2023-10-02 11:41:03.602233 +02:00 < access-control-expose-headers: X-Sentry-Error, Retry-After
  DEBUG   2023-10-02 11:41:03.602243 +02:00 < access-control-allow-origin: *
  DEBUG   2023-10-02 11:41:03.602252 +02:00 < x-sentry-rate-limit-remaining: 39
  DEBUG   2023-10-02 11:41:03.602261 +02:00 < x-sentry-rate-limit-limit: 40
  DEBUG   2023-10-02 11:41:03.602270 +02:00 < x-sentry-rate-limit-reset: 1696239664
  DEBUG   2023-10-02 11:41:03.602280 +02:00 < x-sentry-rate-limit-concurrentremaining: 24
  DEBUG   2023-10-02 11:41:03.602288 +02:00 < x-sentry-rate-limit-concurrentlimit: 25
  DEBUG   2023-10-02 11:41:03.602299 +02:00 < vary: Accept-Language, Cookie
  DEBUG   2023-10-02 11:41:03.602308 +02:00 < content-language: en
  DEBUG   2023-10-02 11:41:03.602317 +02:00 < x-frame-options: deny
  DEBUG   2023-10-02 11:41:03.602326 +02:00 < x-content-type-options: nosniff
  DEBUG   2023-10-02 11:41:03.602335 +02:00 < x-xss-protection: 1; mode=block
  DEBUG   2023-10-02 11:41:03.602347 +02:00 < content-security-policy: font-src * data:; script-src 'self' 'unsafe-inline' 'report-sample' 'unsafe-eval' s1.sentry-cdn.com js.sentry-cdn.com browser.sentry-cdn.com statuspage-production.s3.amazonaws.com static.zdassets.com aui-cdn.atlassian.com connect-cdn.atl-paas.net js.stripe.com ssl.google-analytics.com 'strict-dynamic' cdn.pendo.io data.pendo.io pendo-io-static.storage.googleapis.com pendo-static-5634074999128064.storage.googleapis.com; frame-ancestors 'self' *.sentry.io; img-src * blob: data:; base-uri 'none'; connect-src *; default-src *; style-src * 'unsafe-inline'; object-src 'self'; report-uri https://o1.ingest.sentry.io/api/54785/security/?sentry_key=f724a8a027db45f5b21507e7142ff78e&sentry_release=003b50fb188c99fa08ca3dcd157025bcf5cce190
  DEBUG   2023-10-02 11:41:03.602361 +02:00 < x-envoy-attempt-count: 1
  DEBUG   2023-10-02 11:41:03.602370 +02:00 < x-envoy-upstream-service-time: 31
  DEBUG   2023-10-02 11:41:03.602379 +02:00 < x-served-by: getsentry-web-default-common-production-6f8686b99c-hqk2s
  DEBUG   2023-10-02 11:41:03.602388 +02:00 < x-served-by: frontend-default-59db7b7454-4jwpf
  DEBUG   2023-10-02 11:41:03.602397 +02:00 < strict-transport-security: max-age=31536000; includeSubDomains; preload
  DEBUG   2023-10-02 11:41:03.602406 +02:00 < via: 1.1 google
  DEBUG   2023-10-02 11:41:03.602415 +02:00 < Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
  DEBUG   2023-10-02 11:41:03.602443 +02:00 response status: 200
  DEBUG   2023-10-02 11:41:03.602471 +02:00 body: {"url":"organizations/test-32ecaf969/chunk-upload/","chunkSize":8388608,"chunksPerRequest":64,"maxFileSize":2147483648,"maxRequestSize":33554432,"concurrency":8,"hashAlgorithm":"sha1","compression":["gzip"],"accept":["debug_files","release_files","pdbs","sources","bcsymbolmaps","il2cpp","portablepdbs","artifact_bundles","artifact_bundles_v2"]}
> Bundled 2 files for upload
> Bundle ID: e5000be2-5dac-5409-b7cd-909d2c855ce9
  DEBUG   2023-10-02 11:41:03.948854 +02:00 request POST https://sentry.io/api/0/organizations/test-32ecaf969/artifactbundle/assemble/
  DEBUG   2023-10-02 11:41:03.948941 +02:00 using token authentication
  DEBUG   2023-10-02 11:41:03.949002 +02:00 json body: {"checksum":"9c98490c8d8b95ae876d458a90cc5f30f5d3ebd0","chunks":["9c98490c8d8b95ae876d458a90cc5f30f5d3ebd0"],"projects":["sentry-test"],"version":"com.sentrytestlogger@1.0+1","dist":"1"}
  DEBUG   2023-10-02 11:41:03.949023 +02:00 retry number 0, max retries: 5
  DEBUG   2023-10-02 11:41:03.949286 +02:00 > POST /api/0/organizations/test-32ecaf969/artifactbundle/assemble/ HTTP/1.1
  DEBUG   2023-10-02 11:41:03.949300 +02:00 > Host: sentry.io
  DEBUG   2023-10-02 11:41:03.949309 +02:00 > Accept: */*
  DEBUG   2023-10-02 11:41:03.949317 +02:00 > Connection: TE
  DEBUG   2023-10-02 11:41:03.949324 +02:00 > TE: gzip
  DEBUG   2023-10-02 11:41:03.949332 +02:00 > User-Agent: sentry-cli/2.21.1
  DEBUG   2023-10-02 11:41:03.949374 +02:00 > Authorization: Bearer sntrys_e***
  DEBUG   2023-10-02 11:41:03.949385 +02:00 > Content-Type: application/json
  DEBUG   2023-10-02 11:41:03.949393 +02:00 > Content-Length: 186
  DEBUG   2023-10-02 11:41:04.539260 +02:00 < HTTP/1.1 200 OK
  DEBUG   2023-10-02 11:41:04.539291 +02:00 < server: nginx
  DEBUG   2023-10-02 11:41:04.539342 +02:00 < date: Mon, 02 Oct 2023 09:41:04 GMT
> Nothing to upload, all files are on the server
  DEBUG   2023-10-02 11:41:04.539355 +02:00 < content-type: application/json
  DEBUG   2023-10-02 11:41:04.539363 +02:00 < Content-Length: 38
  DEBUG   2023-10-02 11:41:04.539369 +02:00 < allow: POST, OPTIONS
  DEBUG   2023-10-02 11:41:04.539375 +02:00 < access-control-allow-methods: POST, OPTIONS
  DEBUG   2023-10-02 11:41:04.539384 +02:00 < access-control-allow-headers: X-Sentry-Auth, X-Requested-With, Origin, Accept, Content-Type, Authentication, Authorization, Content-Encoding, sentry-trace, baggage, X-CSRFToken
  DEBUG   2023-10-02 11:41:04.539390 +02:00 < access-control-expose-headers: X-Sentry-Error, Retry-After
  DEBUG   2023-10-02 11:41:04.539397 +02:00 < access-control-allow-origin: *
  DEBUG   2023-10-02 11:41:04.539403 +02:00 < x-sentry-rate-limit-remaining: 39
  DEBUG   2023-10-02 11:41:04.539410 +02:00 < x-sentry-rate-limit-limit: 40
  DEBUG   2023-10-02 11:41:04.539473 +02:00 < x-sentry-rate-limit-reset: 1696239665
  DEBUG   2023-10-02 11:41:04.539497 +02:00 < x-sentry-rate-limit-concurrentremaining: 24
  DEBUG   2023-10-02 11:41:04.539525 +02:00 < x-sentry-rate-limit-concurrentlimit: 25
  DEBUG   2023-10-02 11:41:04.539535 +02:00 < vary: Accept-Language, Cookie
  DEBUG   2023-10-02 11:41:04.539541 +02:00 < content-language: en
  DEBUG   2023-10-02 11:41:04.539548 +02:00 < x-frame-options: deny
  DEBUG   2023-10-02 11:41:04.539554 +02:00 < x-content-type-options: nosniff
  DEBUG   2023-10-02 11:41:04.539560 +02:00 < x-xss-protection: 1; mode=block
  DEBUG   2023-10-02 11:41:04.539566 +02:00 < content-security-policy: base-uri 'none'; object-src 'self'; style-src * 'unsafe-inline'; frame-ancestors 'self' *.sentry.io; script-src 'self' 'unsafe-inline' 'report-sample' 'unsafe-eval' s1.sentry-cdn.com js.sentry-cdn.com browser.sentry-cdn.com statuspage-production.s3.amazonaws.com static.zdassets.com aui-cdn.atlassian.com connect-cdn.atl-paas.net js.stripe.com ssl.google-analytics.com 'strict-dynamic' cdn.pendo.io data.pendo.io pendo-io-static.storage.googleapis.com pendo-static-5634074999128064.storage.googleapis.com; connect-src *; font-src * data:; default-src *; img-src * blob: data:; report-uri https://o1.ingest.sentry.io/api/54785/security/?sentry_key=f724a8a027db45f5b21507e7142ff78e&sentry_release=003b50fb188c99fa08ca3dcd157025bcf5cce190
  DEBUG   2023-10-02 11:41:04.539581 +02:00 < x-envoy-attempt-count: 1
  DEBUG   2023-10-02 11:41:04.539587 +02:00 < x-envoy-upstream-service-time: 84
  DEBUG   2023-10-02 11:41:04.539593 +02:00 < x-served-by: getsentry-web-default-common-production-6f8686b99c-bbtht
  DEBUG   2023-10-02 11:41:04.539598 +02:00 < x-served-by: frontend-default-59db7b7454-shlh6
  DEBUG   2023-10-02 11:41:04.539604 +02:00 < strict-transport-security: max-age=31536000; includeSubDomains; preload
  DEBUG   2023-10-02 11:41:04.539609 +02:00 < via: 1.1 google
  DEBUG   2023-10-02 11:41:04.539615 +02:00 < Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
  DEBUG   2023-10-02 11:41:04.539639 +02:00 response status: 200
  DEBUG   2023-10-02 11:41:04.539647 +02:00 body: {"state":"created","missingChunks":[]}
> File processing complete
> Organization: test-32ecaf969
> Project: sentry-test
> Release: com.sentrytestlogger@1.0+1
> Dist: 1
> Upload type: artifact bundle

Source Map Upload Report
  Minified Scripts
    ~/index.android.bundle (sourcemap at index.android.bundle.map)
  Source Maps
    ~/index.android.bundle.map

> Task :app:createBundleReleaseJsAndAssets_SentryCollectModules_com.sentrytestlogger@1.0+1_1
Sentry Logger [info]: Reading source map from /Users/johndoe/Documents/repos/sentry-test-logger/android/app/build/generated/sourcemaps/react/release/index.android.bundle.map
Sentry Logger [info]: Saving modules to /Users/johndoe/Documents/repos/sentry-test-logger/android/app/src/main/assets/modules.json
Sentry Logger [info]: Resolving modules from paths /Users/johndoe/Documents/repos/sentry-test-logger/android/app/src/main/assets/modules.json
Sentry Logger [info]: Modules collected and saved to: /Users/johndoe/Documents/repos/sentry-test-logger/android/app/src/main/assets/modules.json

> Task :app:installRelease
Installing APK 'app-release.apk' on 'IN2023 - 13' for :app:release
Installed on 1 device.

BUILD SUCCESSFUL in 13s
66 actionable tasks: 6 executed, 60 up-to-date
```

**Comment by loewenheim** [comment]
Thank you for the report, I'll take a look tomorrow.

**Comment by loewenheim** [comment]
Are the files you deleted and are trying to reupload changed at all?

**Comment by jenskuhrjorgensen** [comment]
They have not necessarily changed no.

**Comment by loewenheim** [comment]
If they haven't changed, I actually think this is working as intended—the files still exist in the backend somewhere, so if you want to assemble them into a bundle again, you don't actually need to reupload anything. Does this cause you any actual problems?

**Comment by jenskuhrjorgensen** [comment]
@loewenheim no problems except my mind went through this:
1. Deleted all uploaded bundles from sentry.io
2. Ran the upload command
3. Was told that there was "Nothing to upload, all files are on the server"
4. An artifact bundle was now present at sentry.io
5. 🤯

I just wanted to be 100% sure I was creating, uploading and testing from a clean slate, and the message and behaviour was confusing me.

**Comment by loewenheim** [comment]
Yeah, I can see how this is confusing. I'll think about how the output can be clarified to explain this better.
